src/main/java/frc/robot/subsystems/SwerveWheelModuleSubsystem.java

```diff
@@ -0,0 +1,141 @@
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+//import com.ctre.phoenix.sensors.AbsoluteSensorRange;
+
+//import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+//import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+//import edu.wpi.first.wpilibj.shuffleboard.SimpleWidget;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.util.sendable.SendableRegistry;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.util.MathUtil;
+import frc.robot.util.Constants;
+
+public class SwerveWheelModuleSubsystem extends SubsystemBase {
+    // private final double P = .008;
+    // private final double I = .00001;
+    private final double P = .009;
+    private final double I = .00002;
+
+    private TalonFX angleMotor;
+    private TalonFX speedMotor;
+    private PIDController pidController;
+    private CANcoder angleEncoder;
+    // private boolean calibrateMode;
+    private double encoderOffset;
+    private String motorName;
+    // private SimpleWidget speedLim;
+
+    public SwerveWheelModuleSubsystem(int angleMotorChannel, int speedMotorChannel, int angleEncoderChannel,
+            String motorName, double offset) {
+        // We're using TalonFX motors on CAN.
+        this.angleMotor = new TalonFX(angleMotorChannel);
+        this.speedMotor = new TalonFX(speedMotorChannel);
+        this.angleEncoder = new CANcoder(angleEncoderChannel); // CANCoder Encoder
+        this.speedMotor.setNeutralMode(NeutralModeValue.Coast);
+        // angleMotor.configSupplyCurrentLimit(Constants.kdriveCurrentLimit);
+        // speedMotor.configSupplyCurrentLimit(Constants.kdriveCurrentLimit);
+        this.motorName = motorName;
+
+        this.pidController = new PIDController(P, I, 0); // This is the PID constant,
+        // we're not using any
+        // Integral/Derivative control but increasing the P value will make
+        // the motors more aggressive to changing to angles.
+
+        pidController.enableContinuousInput(0, 360); // This makes the PID controller
+        // understand the fact that for
+        // our setup, 360 degrees is the same as 0 since the wheel loops.
+
+        SendableRegistry.addChild(this, angleMotor);
+        SendableRegistry.addChild(this, speedMotor);
+        SendableRegistry.addChild(this, angleEncoder);
+        SendableRegistry.addLW(this, "Swerve Wheel Module");
+
+        encoderOffset = offset;
+        resetSensor();
+    }
+
+    public void drive(double speed, double angle) {
+        double currentEncoderValue = getPosition();
+        int reverse = setAngle(angle, currentEncoderValue);
+        setSpeed(speed * reverse);
+
+        SmartDashboard.putNumber("Swerve CANCoder " + motorName, getPosition());

```

Comment:
position is also being posted here rn, but this only runs while the teleop is activated. moving this to periodic will ensure it sends position data to dashboard while the robot is disabled.